src/components/structures/TabbedView.tsx

```diff
@@ -54,39 +53,22 @@ export enum TabLocation {
 
 interface IProps<T extends string> {
     tabs: NonEmptyArray<Tab<T>>;
-    initialTabId?: T;
+    activeTabId: T;
     tabLocation?: TabLocation;
-    onChange?: (tabId: T) => void;
+    onChange: (tabId: T) => void;
     screenName?: ScreenName;
 }
 
 export default function TabbedView<T extends string>(props: IProps<T>): JSX.Element {
     const tabLocation = props.tabLocation ?? TabLocation.LEFT;
 
-    const [activeTabId, setActiveTabId] = React.useState<T>((): T => {
-        const initialTabIdIsValid = props.tabs.find((tab) => tab.id === props.initialTabId);
-        // unfortunately typescript doesn't infer the types coorectly if the null check is included above
-        return initialTabIdIsValid && props.initialTabId ? props.initialTabId : props.tabs[0].id;
-    });
+    const activeTabIdPropIsValid = props.tabs.find((tab) => tab.id === props.activeTabId);
+    const activeTabId = activeTabIdPropIsValid ? props.activeTabId : props.tabs[0].id;
```

Comment:
This behaviour feels really unexpected. If you pass an invalid active tab it shouldn't just default to the first IMO. Maybe the caller should pass the activeTab instead to avoid this odd implicit behaviour

Reply:
I agree this is a bit weird, I didn't want to risk changing too much. Do you mean passing the whole active tab object? This feels weirder IMO because you'd still have to pass an array of all the tab objects so it can render the tabs themselves. My vote would be to fail loudly, ie. log an error and be blank, or even throw an exception to cause a soft crash.

Reply:
You can probably protect it via types also, the generics could express that the value has to be one in the provided array

Reply:
Removed then: I think it already is, as illustrated by the fact that I had to add the invalid tab ID as one of the possible Tab IDs in the test I just removed.